fetch.js

```diff
@@ -311,7 +311,13 @@
       xhr.onload = function() {
         var status = (xhr.status === 1223) ? 204 : xhr.status
         if (status < 100 || status > 599) {
-          reject(new TypeError('Network request failed'))
+          var msg = 'Network request failed'
+          if (xhr.responseText) {
+            msg += ' ("' + xhr.responseText + '")'
+          } else {
+            msg += ' (no other details)'
```

Comment:
That part is not as important as providing the other info. I could update the code to omit that else part (although it's nice to let the user know that info as well).